packages/editor/src/components/provider/index.js

```diff
@@ -24,22 +26,192 @@ import { unlock } from '../../lock-unlock';
 const { ExperimentalBlockEditorProvider } = unlock( blockEditorPrivateApis );
 const { PatternsMenuItems } = unlock( editPatternsPrivateApis );
 
+const noop = () => {};
+export const PAGE_CONTENT_BLOCK_TYPES = [
+	'core/post-title',
+	'core/post-featured-image',
+	'core/post-content',
+];
+
+/**
+ * For the Navigation block editor, we need to force the block editor to contentOnly for that block.
+ *
+ * Set block editing mode to contentOnly when entering Navigation focus mode.
+ * this ensures that non-content controls on the block will be hidden and thus
+ * the user can focus on editing the Navigation Menu content only.
+ *
+ * @param {string} navigationBlockClientId ClientId.
+ */
+function useForceFocusModeForNavigation( navigationBlockClientId ) {
+	const { setBlockEditingMode, unsetBlockEditingMode } =
+		useDispatch( blockEditorStore );
+
+	useEffect( () => {
+		if ( ! navigationBlockClientId ) {
+			return;
+		}
+
+		setBlockEditingMode( navigationBlockClientId, 'contentOnly' );
+
+		return () => {
+			unsetBlockEditingMode( navigationBlockClientId );
+		};
+	}, [
+		navigationBlockClientId,
+		unsetBlockEditingMode,
+		setBlockEditingMode,
+	] );
+}
+
+/**
+ * Helper method to extract the post content block types from a template.
+ *
+ * @param {Array} blocks Template blocks.
+ *
+ * @return {Array} Flattened object.
+ */
+function extractedPageContentBlockTypesFromTemplateBlocks( blocks ) {
+	const result = [];
+	for ( let i = 0; i < blocks.length; i++ ) {
+		// Since the Query Block could contain PAGE_CONTENT_BLOCK_TYPES block types,
+		// we skip it because we only want to render stand-alone page content blocks in the block list.
+		if ( [ 'core/query' ].includes( blocks[ i ].name ) ) {
+			continue;
+		}
+		if ( PAGE_CONTENT_BLOCK_TYPES.includes( blocks[ i ].name ) ) {
+			result.push( createBlock( blocks[ i ].name ) );
+		}
+		result.push(
+			...extractedPageContentBlockTypesFromTemplateBlocks(
+				blocks[ i ].innerBlocks
+			)
+		);
```

Comment:
```suggestion
		if ( blocks[ i ].innerBlocks.length ) {
			result.push(
				...extractedPageContentBlockTypesFromTemplateBlocks(
					blocks[ i ].innerBlocks
				)
			);
		}
```

Might save 0.000001 of a nanosecond 😄 and prevent pushing empty arrays to results